### PR TITLE
convert LdapExceptions to checked DaoExceptions instead of unchecked …

### PR DESCRIPTION
**Pull request by frett**
…RuntimeExceptions

### CODE REVIEW

**Comment by frett** [comment]
@hlbraddock

File: ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java [hunk 1]
```diff
@@ -458,7 +456,7 @@ private void modifyGroupMembership(User user, Group group, AttributeModification
             modifyEntry(conn, groupDn, attributeModificationType, userDn, LDAP_ATTR_MEMBER);
 
         } catch (final LdapException e) {
-            throw Throwables.propagate(e);
+            throw convertLdapException(e);
```
Comment (hlbraddock) [inline ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java]: Why not simply `throw new LdaptiveDaoException(e);`?

**Comment by hlbraddock** [comment]
+1 This looks good (aside from my one comment/question).

File: ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java (hunk 1 above)
Comment (frett) [inline ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java]: down the line I want to try and actually classify the exception being thrown, so I centralized the simple conversion logic to make it easier to parse and convert more intelligently in the future.

**Pull request merged**